src/postgres/commands/PostgresAccountWizard/PostgresServerNameStep.ts

```diff
@@ -43,11 +42,18 @@ async function validatePostgresServerName(name: string, client: PostgreSQLManage
     }
 
     const availabilityRequest: NameAvailabilityRequest = { name: name, type: "Microsoft.DBforPostgreSQL" };
+    const characterMatch = name.match(/^(?![-])[a-zA-Z0-9-]*(?<![-])$/);
+    const prefixSuffixMatch = name.match(/^[-]*.*[-]$/);
     const availability: NameAvailability = (await client.checkNameAvailability.execute(availabilityRequest));
 
     if (!availability.nameAvailable) {
-
-        return availability.message;
+        if (availability.reason === 'AlreadyExists') {
+            return localize('serverNameAvailabilityCheck', 'Server name "{0}" is not available.', name);
+        } else if (prefixSuffixMatch) {
```

Comment:
This condition wasn't met for a server name starting with a hyphen
<img width="509" alt="Screen Shot 2020-05-07 at 10 35 35 AM" src="https://user-images.githubusercontent.com/22795803/81326322-84517080-904e-11ea-8a2b-319b18f10abd.png">